### ISSUE TITLE
Add FilterableTable component

### ISSUE THREAD

**Issue by tarcinil** [spotify/backstage#315]
There is already a `SortableTable` inside of core components.

I can also see a need for a `FilterableTable` based on the data that has been loaded.  I also looked at Material Table for this.

This is completely dependent on #161 as the project would need to adopt a Datatable standardization.  This could also require porting `SortableTable` so that they don't have multiple APIs.

**Comment by freben** [comment]
I can foresee this being a single `DataTable` shared component that supports both filtering and sorting by default, but each can be enabled or disabled separately.

**Comment by tarcinil** [comment]
I can see that as a viable option, but I would worry about how many configurations that single component could have and how a possible configuration could be missed.

Is it possible to create a `DataTable` component and extend from it to enable `SortableTable` and `FilterableTable`?  This would also make the UX much cleaner rather than configuring _N_ flags under `DataTable`.

**Comment by freben** [comment]
Sure, that would work. I was hoping it would just be two props `{ sortable?: bool, filterable?: bool }` though. And making a wrapper component that hardcodes that could be made too, but it's probably not super helpful (the cognitive load of remembering several components over several files is arguably higher than looking through options, with autocomplete, for a single component). And what about `SortableAndFilterableTable`? :) But let's play around and make an API that is comfortable and straightforward!

**Comment by freben** [comment]
we've also discussed maybe adding this wholesale https://github.com/tannerlinsley/react-table

**Comment by tarcinil** [comment]
I like that library and it would allow for a spec that is completely managed inside the project.

I can see your point about the cognitive load.  I am just concerned with adoptability vs maintainability.  I think my concern can be handled well with enough documentation though.

**Comment by stefanalund** [comment]
Ideally we would have a recommended grid component. A first step (until we have a [Storybook](https://github.com/spotify/backstage/issues/255) set up) could be to replace the default Material-UI Table in the ExampleComponent that the CLI creates.

@tarcinil would you be up for taking a stab at that?

**Comment by Rugvip** [comment]
I think this discussion has gone into a duplicate of #161, let's close this and continue there.
